src/extension/wps/src/main/java/au/org/emii/wps/gogoduck/index/FeatureSourceIndexReader.java

```diff
@@ -40,24 +40,29 @@ public FeatureSourceIndexReader(Catalog catalog, Map<String, String> urlSubstitu
     @Override
     public Set<DownloadRequest> getDownloadList(String typeName, String timeField, String sizeField, String urlField, SubsetParameters subset) throws GoGoDuckException {
         Set<DownloadRequest> result = new LinkedHashSet<>();
+        Query query = new Query();
 
-        String timeCoverageStart = subset.getTimeRange().getStart().toString();
-        String timeCoverageEnd = subset.getTimeRange().getEnd().toString();
+        if (subset.getTimeRange() != null) {
+            String timeCoverageStart = subset.getTimeRange().getStart().toString();
+            String timeCoverageEnd = subset.getTimeRange().getEnd().toString();
+            Filter cqlFilter;
 
-        Filter cqlFilter;
-
-        try {
-            cqlFilter = CQL.toFilter(
-                    String.format("%s >= '%s' AND %s <= '%s'",
-                            timeField, timeCoverageStart,
-                            timeField, timeCoverageEnd)
-            );
-        } catch (CQLException e) {
-            logger.error(e.getMessage(), e);
-            throw new GoGoDuckException(e.getMessage());
+            try {
+                cqlFilter = CQL.toFilter(
+                        String.format("%s >= '%s' AND %s <= '%s'",
+                                timeField, timeCoverageStart,
+                                timeField, timeCoverageEnd)
+                );
+                query = new Query(typeName, cqlFilter, new String[]{urlField, sizeField});
+            } catch (CQLException e) {
+                logger.error(e.getMessage(), e);
+                throw new GoGoDuckException(e.getMessage());
+            }
         }
 
-        Query query = new Query(typeName, cqlFilter, new String[]{urlField, sizeField});
+        if (query == null) {
```

Comment:
This could never be null - you assign to it above - so this code path will never get executed.